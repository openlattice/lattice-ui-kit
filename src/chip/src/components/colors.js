import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
