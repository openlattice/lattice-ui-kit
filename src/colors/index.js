/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
