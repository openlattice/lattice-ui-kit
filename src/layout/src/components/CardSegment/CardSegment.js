// @flow
import styled, { css } from 'styled-components';

type ComputedSegmentProps = {
  bgColor :string;
  noBleed :boolean;
  onClick :() => void;
  padding :'sm' | 'md';
  vertical :boolean;
};

const getSegmentComputedStyles = (props :ComputedSegmentProps) => {
  const {
    bgColor,
    noBleed,
    onClick,
    padding,
    vertical,
  } = props;

  let backgroundColor = 'transparent';
  if (bgColor) {
    backgroundColor = `${bgColor}`;
  }

  let cursor = 'auto';
  if (onClick) {
    cursor = 'pointer';
  }

  const finalMargin = noBleed ? '0 30px' : '0';
  const lrPadding = noBleed ? '0' : '30px';

  let finalPadding = `30px ${lrPadding}`;
  if (padding === 'sm') {
    finalPadding = `10px ${lrPadding}`;
  }
  else if (padding === 'md') {
    finalPadding = `20px ${lrPadding}`;
  }

  let flexDirection = 'row';
  if (vertical) {
    flexDirection = 'column';
  }

  const styles = css`
    background-color: ${backgroundColor};
    flex-direction: ${flexDirection};
    margin: ${finalMargin};
    padding: ${finalPadding};
    &:hover {
      cursor: ${cursor}
    }
  `;

  return styles;
};

const CardSegment = styled.div`
  display: flex;
  flex: 1 0 auto;
  position: relative;
  ${getSegmentComputedStyles}
`;

export default CardSegment;
export {
  getSegmentComputedStyles,
};
