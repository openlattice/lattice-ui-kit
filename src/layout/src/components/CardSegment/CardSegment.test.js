import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import CardSegment from './CardSegment';

import { NEUTRAL } from '../../../../colors';

describe('CardSegment', () => {

  describe('snapshots', () => {

    test('without children', () => {
      const wrapper = mount(<CardSegment />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with children', () => {
      const wrapper = mount((
        <CardSegment>
          <div>child 1</div>
          <div>child 2</div>
        </CardSegment>
      ));
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('render', () => {

    test('should render div', () => {
      const wrapper = mount(<CardSegment />);
      expect(wrapper.find('div')).toHaveLength(1);
    });

    describe('computed styles', () => {

      test('bgColor', () => {
        const wrapper = mount(<CardSegment bgColor="#ffffff" />);
        expect(wrapper).toHaveStyleRule('background-color', '#ffffff');
      });

      test('padding - sm', () => {
        const wrapper = mount(<CardSegment padding="sm" />);
        expect(wrapper).toHaveStyleRule('padding', '10px 30px 10px 30px');
      });

      test('padding - md', () => {
        const wrapper = mount(<CardSegment padding="md" />);
        expect(wrapper).toHaveStyleRule('padding', '20px 30px 20px 30px');
      });

      test('vertical', () => {
        const wrapper = mount(<CardSegment vertical />);
        expect(wrapper).toHaveStyleRule('flex-direction', 'column');
      });

    });

  });

  describe('props', () => {

    describe('borderless', () => {

      test('should set border-bottom with "borderless=false"', () => {
        const wrapper = mount(<CardSegment />);
        expect(wrapper).toHaveStyleRule('border-bottom', `1px solid ${NEUTRAL.N100}`);
      });

      test('should not set border-bottom with "borderless=true"', () => {
        const wrapper = mount(<CardSegment borderless />);
        expect(wrapper).not.toHaveStyleRule('border-bottom');
      });

    });

    describe('indent', () => {

      test('should indent via padding by default', () => {
        const wrapper = mount(<CardSegment indent={10} />);
        expect(wrapper).toHaveStyleRule('margin', '0');
        expect(wrapper).toHaveStyleRule('padding', '30px 30px 30px 130px');
      });

      test('should indent via margin with "noBleed=true"', () => {
        const wrapper = mount(<CardSegment indent={10} noBleed />);
        expect(wrapper).toHaveStyleRule('margin', '0 30px 0 130px');
        expect(wrapper).toHaveStyleRule('padding', '30px 0 30px 0');
      });

      test('should respect top/bottom padding with "padding=sm"', () => {
        const wrapper = mount(<CardSegment indent={10} padding="sm" />);
        expect(wrapper).toHaveStyleRule('margin', '0');
        expect(wrapper).toHaveStyleRule('padding', '10px 30px 10px 130px');
      });

      test('should respect top/bottom padding with "padding=md"', () => {
        const wrapper = mount(<CardSegment indent={10} padding="md" />);
        expect(wrapper).toHaveStyleRule('margin', '0');
        expect(wrapper).toHaveStyleRule('padding', '20px 30px 20px 130px');
      });

      test('should ignore negative values', () => {
        const wrapper = mount(<CardSegment indent={-10} />);
        expect(wrapper).toHaveStyleRule('margin', '0');
        expect(wrapper).toHaveStyleRule('padding', '30px 30px 30px 30px');
      });

    });

    describe('noBleed', () => {

      test('should use only padding with "noBleed=false"', () => {
        const wrapper = mount(<CardSegment />);
        expect(wrapper).toHaveStyleRule('margin', '0');
        expect(wrapper).toHaveStyleRule('padding', '30px 30px 30px 30px');
      });

      test('should use left/right margin instead of padding with "noBleed=true"', () => {
        const wrapper = mount(<CardSegment noBleed />);
        expect(wrapper).toHaveStyleRule('margin', '0 30px 0 30px');
        expect(wrapper).toHaveStyleRule('padding', '30px 0 30px 0');
      });

    });

    describe('onClick', () => {

      test('should have "cursor: inherit" by default for hover', () => {
        const wrapper = mount(<CardSegment />);
        expect(wrapper).toHaveStyleRule('cursor', 'inherit', { modifier: ':hover' });
      });

      test('should have "cursor: pointer" for hover', () => {
        const wrapper = mount(<CardSegment onClick={() => {}} />);
        expect(wrapper).toHaveStyleRule('cursor', 'pointer', { modifier: ':hover' });
      });

    });

    describe('padding', () => {

      test('should set padding directly', () => {
        const wrapper = mount(<CardSegment padding="10px 20px" />);
        expect(wrapper).toHaveStyleRule('padding', '10px 20px');
      });

      test('should override indent', () => {
        const wrapper = mount(<CardSegment indent={10} padding="50px" />);
        expect(wrapper).toHaveStyleRule('padding', '50px');
      });

      test('should respect margin with "noBleed=true"', () => {
        const wrapper = mount(<CardSegment noBleed padding="50px" />);
        expect(wrapper).toHaveStyleRule('margin', '0 30px 0 30px');
        expect(wrapper).toHaveStyleRule('padding', '50px');
      });

    });

  });

});
