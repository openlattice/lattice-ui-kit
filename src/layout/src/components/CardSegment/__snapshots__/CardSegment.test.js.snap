// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSegment snapshots with children 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  position: relative;
  background-color: transparent;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 30px 30px 30px 30px;
}

.c1:hover {
  cursor: inherit;
}

.c1 .c0 {
  border-bottom: 1px solid #dcdce7;
}

.c1 .c0:first-child {
  border-radius: 3px 3px 0 0;
}

.c1 .c0:last-child {
  border-bottom: 0;
  border-radius: 0 0 3px 3px;
}

<CardSegment>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "c0",
          "isStatic": false,
          "lastClassName": "c1",
          "rules": Array [
            "display:flex;flex:1 0 auto;position:relative;",
            [Function],
            " & &{border-bottom:1px solid ",
            "#dcdce7",
            ";}& &:first-child{border-radius:3px 3px 0 0;}& &:last-child{border-bottom:0;border-radius:0 0 3px 3px;}",
          ],
        },
        "displayName": "CardSegment",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "c0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0 c1"
    >
      <div>
        child 1
      </div>
      <div>
        child 2
      </div>
    </div>
  </StyledComponent>
</CardSegment>
`;

exports[`CardSegment snapshots without children 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  position: relative;
  background-color: transparent;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 30px 30px 30px 30px;
}

.c1:hover {
  cursor: inherit;
}

.c1 .c0 {
  border-bottom: 1px solid #dcdce7;
}

.c1 .c0:first-child {
  border-radius: 3px 3px 0 0;
}

.c1 .c0:last-child {
  border-bottom: 0;
  border-radius: 0 0 3px 3px;
}

<CardSegment>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "c0",
          "isStatic": false,
          "lastClassName": "c1",
          "rules": Array [
            "display:flex;flex:1 0 auto;position:relative;",
            [Function],
            " & &{border-bottom:1px solid ",
            "#dcdce7",
            ";}& &:first-child{border-radius:3px 3px 0 0;}& &:last-child{border-bottom:0;border-radius:0 0 3px 3px;}",
          ],
        },
        "displayName": "CardSegment",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "c0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0 c1"
    />
  </StyledComponent>
</CardSegment>
`;
